lib/output.c

```diff
@@ -2710,6 +2710,27 @@ grn_output_result_set_open_metadata(grn_ctx *ctx,
     }
     ctx->impl->output.arrow_stream_writer =
       grn_arrow_stream_writer_open(ctx, outbuf);
+    if (ctx->impl->output.arrow_metadata_data_type &&
+        ctx->impl->output.arrow_metadata_label.value) {
+      grn_obj label_buffer;
+      GRN_TEXT_INIT(&label_buffer, 0);
+      GRN_TEXT_SET(ctx,
+                   &label_buffer,
+                   ctx->impl->output.arrow_metadata_label.value,
+                   ctx->impl->output.arrow_metadata_label.length);
+      GRN_TEXT_PUTC(ctx, &label_buffer, '\0');
+      grn_arrow_stream_writer_add_metadata(
+        ctx,
+        ctx->impl->output.arrow_stream_writer,
+        "GROONGA:data_type",
+        ctx->impl->output.arrow_metadata_data_type);
+      grn_arrow_stream_writer_add_metadata(
+        ctx,
+        ctx->impl->output.arrow_stream_writer,
+        "GROONGA:label",
+        GRN_TEXT_VALUE(&label_buffer));
+      GRN_OBJ_FIN(ctx, &label_buffer);
+    }
   }
```

Comment:
Is `#ifdef GRN_WITH_APACHE_ARROW ... #endif` unnecessary in this block?

Reply:
I think we don't need it because we have already checked whether the output format is Apache Arrow format or not.

Reply:
The existed code doesn't handle it at all when it  check the `GRN_CONTENT_APACHE_ARROW`. Do we really need it here? If so, do we also add it in the other places too?

Reply:
We don't need `#ifdef GRN_WITH_APACHE_ARROW` here.

`grn_arrow_stream_writer_*()` are always defined to avoid many `#ifdef GRN_WITH_APACHE_ARROW` in many places.

Reply:
I see.